### PR TITLE
cmd: wrap up tikv config

### PR DESCRIPTION
**Pull request by overvenus**
###  What have you changed?

Add a new Config in the cmd crate, it wraps TiKvConfig, also this PR moves persist feature from the TiKvConfig to the new Config. This change allows adding further configuration from other components.

It's preparation for the upcoming backup feature.

###  What is the type of the changes?

Pick one of the following and delete the others:

- Engineering (engineering change which doesn't change any feature or fix any issue)

###  How is the PR tested?

Integration tests.

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No.

###  Does this PR affect `tidb-ansible`?

No.

### CODE REVIEW

**Comment by 5kbpers** [comment]
Shall we consider to fix the `--version` bug through this PR?

**Comment by overvenus** [comment]
> Shall we consider to fix the --version bug through this PR?

We can fix it in other PR, let's focus on config change.

File: cmd/src/config.rs [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+use tikv::config::TiKvConfig;
+
+const LAST_CONFIG_FILE: &str = "last_tikv.toml";
+
+#[derive(Clone, Serialize, Deserialize, PartialEq, Debug)]
+#[serde(default)]
+#[serde(deny_unknown_fields)]
+#[serde(rename_all = "kebab-case")]
+pub struct Config {
+    #[serde(flatten)]
```
Comment (hicqu) [inline cmd/src/config.rs]: Should we move serde functions from `TiKVConfig` to here? If we add more fields in `tikv.toml`, they belong to `TiKVConfig` or this `Config`? Personally I think it's better to deserialize other component configs from `other_component.toml`, and just combine them into `Config`.

Comment (overvenus) [reply cmd/src/config.rs]: > Should we move serde functions from TiKVConfig to here?

I don't think so, serde functions should stay the same mod as `TikvConfig`. 

> If we add more fields in tikv.toml, they belong to TiKVConfig or this Config? 

It depends, if it belongs to `TikvConfig` then it should stay in `TikvConfig`.

> Personally I think it's better to deserialize other component configs from other_component.toml, 

Well, I don't think it's a good idea to introduce multiple config files.

> and just combine them into Config.

Do you mean the `TikvConfig`?

**Comment by overvenus** [comment]
Close temporarily, I don't have enough PR slots. :(

**Pull request closed without merging**